packages/typescript-eslint/src/config-helper.ts

```diff
@@ -133,12 +157,15 @@ export function config(
         const name = [config.name, extension.name].filter(Boolean).join('__');
         return {
           ...extension,
-          ...(config.files && { files: config.files }),
-          ...(config.ignores && { ignores: config.ignores }),
+          ...Object.fromEntries(
+            Object.entries(config).filter(([key]) =>
+              ['files', 'ignores'].includes(key),
+            ),
+          ),
```

Comment:
Nit: If `files` or `ignores` was invalid but falsy (`0 | '' | undefined | null`), it was silently removed. Instead, we should keep it, and let ESLint handle any invalid values